test/tests/functional/pbs_resv_end_hook_set_attrs.py

```diff
@@ -0,0 +1,61 @@
+from tests.functional import *
+import time
+
+
+class TestSetAttributes(TestFunctional):
+
+    """
+    Testsuite to verify the read privileges in the reservation
+    end hook for an advanced reservation.
+    """
+
+    def setUp(self):
+        """ Create and import the python script for reservation end hook
+        in setup method"""
+
+        super(TestSetAttributes, self).setUp()
+        hook_script = """
+import pbs
+e=pbs.event()
+
+pbs.logmsg(pbs.LOG_DEBUG, 'Reservation End Hook name - %s' % (pbs.event().hook_name))
+
+if e.type == pbs.RESV_END:
+    pbs.event().resv.reserve_id = 10
```

Comment:
Why are you setting the reserve_id of the reservation?  Shouldn't that be read only and set by the server?

Reply:
Yes, the privileges are readonly and server will set it. This test case is to verify the read permissions.